addons/point_of_sale/models/res_config_settings.py

```diff
@@ -1,21 +1,316 @@
+    pos_use_pricelist = fields.Boolean(related='pos_config_id.use_pricelist', readonly=False)
+    pos_warehouse_id = fields.Many2one(related='pos_config_id.warehouse_id', readonly=False, string="Warehouse (PoS)")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # STEP: Remove the 'pos' fields from each vals.
+        #   They will be written atomically to `pos_config_id` after the super call.
+        pos_config_id_to_fields_vals_map = {}
+
+        for vals in vals_list:
+            pos_config_id = vals.get('pos_config_id', False)
+            if not pos_config_id:
+                continue
+
+            pos_fields_vals = {}
+            for _field in self._fields.values():
+                val = vals.get(_field.name, None)
+
+                # Add only to pos_fields_vals if
+                #   1. _field is in vals -- meaning, the _field is in view.
+                #   2. _field has 'pos' attribute
+                if hasattr(_field, 'pos') and val is not None:
+                    pos_fields_vals[getattr(_field, 'pos')] = val
+                    del vals[_field.name]
+                    continue
+
+            if vals.get('pos_cash_rounding', False):
+                vals['group_cash_rounding'] = True
+
+            if vals.get('pos_use_pricelist', False):
+                vals['group_product_pricelist'] = True
+
+            pos_config_id_to_fields_vals_map[pos_config_id] = pos_fields_vals
+
+        # STEP: Call super on the modified vals_list.
+        # NOTE: When creating `res.config.settings` records, it doesn't write on *unmodified* related fields.
+        result = super().create(vals_list)
+
+        # STEP: Finally, we write the value of 'pos' fields to 'pos_config_id'.
+        # TODO-JCB: Why are there failing tests saying `pos.config` is not in .env?
+        if 'pos.config' in self.env:
+            for pos_config_id, pos_fields_vals in pos_config_id_to_fields_vals_map.items():
+                pos_config = self.env['pos.config'].browse(pos_config_id)
+                pos_config.write(pos_fields_vals)
+
+        return result
 
     def set_values(self):
         super(ResConfigSettings, self).set_values()
         if not self.group_product_pricelist:
             self.env['pos.config'].search([
                 ('use_pricelist', '=', True)
             ]).use_pricelist = False
+
+        if not self.group_cash_rounding:
+            self.env['pos.config'].search([
+                ('cash_rounding', '=', True)
+            ]).cash_rounding = False
+
+    def action_pos_config_create_new(self):
+        return {
+            'view_mode': 'form',
+            'view_id': self.env.ref('point_of_sale.pos_config_view_form').id,
+            'res_model': 'pos.config',
+            'type': 'ir.actions.act_window',
+            'target': 'new',
+            'res_id': False,
+            'context': {'pos_config_open_modal': True, 'pos_config_basic': True},
```

Comment:
If I've well understood, it for when we are creating a new pos config from the res_config_settings.

Do you think we can name it "pos_config_create_mode" or something like this to be more explicit ?